emails/views.py

```diff
@@ -393,22 +384,50 @@ def _sns_message(message_json):
     return response
 
 
-def _handle_reply(message_json):
-    mail = message_json['mail']
-    in_reply_to = None
-    for header in mail['headers']:
+def _get_keys_from_headers(headers):
+    for header in headers:
         if header['name'].lower() == 'in-reply-to':
             in_reply_to = header['value']
             message_id_bytes = get_message_id_bytes(in_reply_to)
     if in_reply_to is None:
         incr_if_enabled('mail_to_replies_without_in_reply_to', 1)
         raise Exception("Received mail to reply address without In-Reply-To")
-    (lookup_key, encryption_key) = derive_reply_keys(
-        message_id_bytes
-    )
+    return derive_reply_keys(message_id_bytes)
+
+
+def _get_reply_record_from_lookup_key(lookup_key):
     lookup = b64_lookup_key(lookup_key)
-    reply_record = Reply.objects.get(lookup=lookup)
-    address = reply_record.relay_address or reply_record.domain_address
+    return Reply.objects.get(lookup=lookup)
+
+
+def _strip_localpart_tag(address):
+    [localpart, domain] = address.split('@')
+    subaddress_parts = localpart.split('+')
```

Comment:
As far as I know no spec defines that the `+` part is stripped from email addresses - it's just a convention used by e.g. Gmail. In other words, it's perfectly possible that for some email providers, email addresses with a `+` and the same prefix before the `+` should go to different people.

In practice I suppose it'll probably be fine, but I thought it couldn't harm to call it out :)

Reply:
Hmm ... good point. And the other complication is that we don't know which domains are "backed" by an email provider like Gmail that supports localpart tags.

After walking thru how this affects this reply code if an email provider `example.com` treats `user123` and `user123+456` as *different* email addresses ...

This will break `user123+456@example.com`'s replies if:
* `user123+456@example.com` IS a premium relay user
* `user123@example.com` is also a relay user
* `user123@example.com` is NOT a premium user

So, I updated the code to check if the `From` matches either exactly OR with the localpart tag stripped off.

Good catch!